src/matlab2tikz.m

```diff
@@ -2414,12 +2416,24 @@ case size(cData,2) % specific y-data is given
     % to avoid it for the most default situations, e.g., when dx and dy are
     % rank-1-matrices.
     % Enforce 'z buffer=sort' if shader is flat
-    if isempty(strfind(opts_get(opts, 'shader'),'interp')) ||...
-        any(~isnan(dx(1,:)) & dx(1,:) ~= dx(2,:)) ||...
-        any(~isnan(dy(:,1)) & dy(:,1) ~= dy(:,2))
+    % Avoid 'z buffer=sort' for hist3D plots
+    isShaderFlat = isempty(strfind(opts_get(opts, 'shader'),'interp'));
+    isHist3D     = strcmpi(get(handle,'tag'),'hist3');
+    if m2t.axesContainers{end}.is3D && isShaderFlat && ~isHist3D
```

Comment:
Fixes a regression in 3D bar introduced when enforcing teh zbuffer sort with flat shader.